docs/AsyncMoya.md

```diff
@@ -0,0 +1,68 @@
+# Async/await
+
+Moya provides an optional `Async/await` implementation of
+`MoyaProvider` that does a few interesting things. Instead of
+calling the `request()` method and providing a callback closure
+to be executed when the request completes, we use `await`.
+
+```swift
+let provider = MoyaProvider<GitHub>()
+```
+
+After that simple setup, you're off to the races:
+
+```swift
+AsyncTask {
+    let result = await provider.asyncRequest(.zen) //return type `Result<Response, MoyaError>`
+    switch result {
+    case let .success(response):
+        // do something with the data
+    case let .failure(error):
+        // handle the error
+    default:
+        break
```

Comment:
I don't think we need this `default` case as `Result` should be `@frozen`, not accepting any new cases in its enum.

Reply:
removed